pdns/lwres.cc

```diff
@@ -327,12 +327,20 @@ LWResult::Result asyncresolve(const ComboAddress& ip, const DNSName& domain, int
       Socket s(ip.sin4.sin_family, SOCK_STREAM);
 
       s.setNonBlocking();
-      ComboAddress local = pdns::getQueryLocalAddress(ip.sin4.sin_family, 0);
+      if (SyncRes::s_tcp_fast_open_connect) {
+        try {
+          s.setFastOpenConnect();
+        }
+        catch (const NetworkError& e) {
+          g_log << Logger::Error << "tcp-fast-connect enabled but returned error: " << e.what() << endl;
```

Comment:
I'm a bit afraid of flooding the logs with that.. Perhaps we could do that test at startup by creating a socket and trying to enable Fast Open Connect? If that works we close the socket right away, if it doesn't we log _once_.